rule_stop.go

```diff
@@ -20,8 +27,31 @@ func (slr stopLossRule) IsSatisfied(index int, record *TradingRecord) bool {
 	if !record.CurrentPosition().IsOpen() {
 		return false
 	}
-
-	openPrice := record.CurrentPosition().CostBasis()
+	openPrice := record.CurrentPosition().EntranceOrder().Price
 	loss := slr.Indicator.Calculate(index).Div(openPrice).Sub(big.ONE)
+	if record.currentPosition.IsShort() {
```

Comment:
try to avoid using the private fields when you can. Here you should replace `record.currentPosition` with `record.CurrentPosition()`